packages/core/src/lib/typescript-schema-printer.ts

```diff
@@ -1,106 +1,60 @@
 import {
+  type GraphQLNamedType,
   type GraphQLSchema,
-  GraphQLScalarType,
-  GraphQLEnumType,
   type GraphQLType,
-  GraphQLNonNull,
-  type GraphQLNamedType,
-  GraphQLList,
+  GraphQLEnumType,
   GraphQLInputObjectType,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLScalarType,
   introspectionTypes,
 } from 'graphql'
 import type { InitialisedList } from './core/initialise-lists'
 
 const introspectionTypesSet = new Set(introspectionTypes)
 
+const SCALARS = {
+  ID: 'string',
+  Boolean: 'boolean',
+  String: 'string',
+  Int: 'number',
+  Float: 'number',
+  JSON: `import('@keystone-6/core/types').JSONValue`,
+  Decimal: `import('@keystone-6/core/types').Decimal | string`,
+} as const
+
 function stringify (x: string) {
   return JSON.stringify(x).slice(1, -1)
 }
 
-function printEnumTypeDefinition (type: GraphQLEnumType) {
-  return [
-    `export type ${type.name} =`,
-    type
-      .getValues()
-      .map(x => `  | '${stringify(x.name)}'`)
-      .join('\n'),
-  ].join('\n')
-}
-
-function printTypeReference (type: GraphQLType, scalars: Record<string, string>): string {
-  if (type instanceof GraphQLNonNull) {
-    return printTypeReferenceWithoutNullable(type.ofType, scalars)
-  }
-  return `${printTypeReferenceWithoutNullable(type, scalars)} | null`
+function printTypeReference (type: GraphQLType): string {
+  if (type instanceof GraphQLNonNull) return printTypeReferenceWithoutNullable(type.ofType)
+  return `${printTypeReferenceWithoutNullable(type)} | null`
 }
 
 function printTypeReferenceWithoutNullable (
   type: GraphQLNamedType | GraphQLList<GraphQLType>,
-  scalars: Record<string, string>
 ): string {
-  if (type instanceof GraphQLList) {
-    return `ReadonlyArray<${printTypeReference(type.ofType, scalars)}> | ${printTypeReference(
-      type.ofType,
-      scalars
-    )}`
-  }
+  if (type instanceof GraphQLList) return `ReadonlyArray<${printTypeReference(type.ofType)}> | ${printTypeReference(type.ofType)}`
 
   const name = type.name
   if (type instanceof GraphQLScalarType) {
-    if (scalars[name] === undefined) return 'any'
-    return `Scalars['${stringify(name)}']` // TODO: inline?
```

Comment:
I don't see why we needed to refer to this `Scalars` object continually for our resolved types; this isn't helpful to developers.